Facing benchmark.h:1050:41: error: variable 'fn' has function type. Even with v1.3.1

```
make
Scanning dependencies of target example
[  4%] Building CXX object example/CMakeFiles/example.dir/example.cpp.o
[  8%] Linking CXX executable example
[  8%] Built target example
Scanning dependencies of target multisink
[ 12%] Building CXX object example/CMakeFiles/multisink.dir/multisink.cpp.o
[ 16%] Linking CXX executable multisink
[ 16%] Built target multisink
Scanning dependencies of target spdlog-utests
[ 20%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_errors.cpp.o
[ 24%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_file_helper.cpp.o
[ 28%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_file_logging.cpp.o
[ 32%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_misc.cpp.o
[ 36%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_pattern_formatter.cpp.o
[ 40%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_async.cpp.o
[ 44%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_registry.cpp.o
[ 48%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_macros.cpp.o
[ 52%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/utils.cpp.o
[ 56%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/main.cpp.o
[ 60%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_mpmc_q.cpp.o
[ 64%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_fmt_helper.cpp.o
[ 68%] Linking CXX executable spdlog-utests
[ 68%] Built target spdlog-utests
Scanning dependencies of target formatter-bench
[ 72%] Building CXX object bench/CMakeFiles/formatter-bench.dir/formatter-bench.cpp.o
In file included from /tmp/tmp.CP8z4tf0ew/vendor/spdlog/bench/formatter-bench.cpp:6:0:
/usr/local/include/benchmark/benchmark.h: In instantiation of 'benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...)::<lambda(benchmark::State&)> [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]':
/usr/local/include/benchmark/benchmark.h:1050:48:   required from 'struct benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...) [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]::<lambda(class benchmark::State&)>'
/usr/local/include/benchmark/benchmark.h:1049:38:   required from 'benchmark::internal::Benchmark* benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...) [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]'
/tmp/tmp.CP8z4tf0ew/vendor/spdlog/bench/formatter-bench.cpp:49:79:   required from here
/usr/local/include/benchmark/benchmark.h:1050:41: error: variable 'fn' has function type
       name, [=](benchmark::State& st) { fn(st, args...); });
                                         ^
compilation terminated due to -Wfatal-errors.
bench/CMakeFiles/formatter-bench.dir/build.make:62: recipe for target 'bench/CMakeFiles/formatter-bench.dir/formatter-bench.cpp.o' failed
make[2]: *** [bench/CMakeFiles/formatter-bench.dir/formatter-bench.cpp.o] Error 1
CMakeFiles/Makefile2:1168: recipe for target 'bench/CMakeFiles/formatter-bench.dir/all' failed
make[1]: *** [bench/CMakeFiles/formatter-bench.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

```
cmake --build _builds --target install
Scanning dependencies of target example
[  4%] Building CXX object example/CMakeFiles/example.dir/example.cpp.o
[  8%] Linking CXX executable example
[  8%] Built target example
Scanning dependencies of target multisink
[ 12%] Building CXX object example/CMakeFiles/multisink.dir/multisink.cpp.o
[ 16%] Linking CXX executable multisink
[ 16%] Built target multisink
Scanning dependencies of target spdlog-utests
[ 20%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_errors.cpp.o
[ 24%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_file_helper.cpp.o
[ 28%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_file_logging.cpp.o
[ 32%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_misc.cpp.o
[ 36%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_pattern_formatter.cpp.o
[ 40%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_async.cpp.o
[ 44%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_registry.cpp.o
[ 48%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_macros.cpp.o
[ 52%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/utils.cpp.o
[ 56%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/main.cpp.o
[ 60%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_mpmc_q.cpp.o
[ 64%] Building CXX object tests/CMakeFiles/spdlog-utests.dir/test_fmt_helper.cpp.o
[ 68%] Linking CXX executable spdlog-utests
[ 68%] Built target spdlog-utests
Scanning dependencies of target formatter-bench
[ 72%] Building CXX object bench/CMakeFiles/formatter-bench.dir/formatter-bench.cpp.o
In file included from /tmp/tmp.CP8z4tf0ew/vendor/spdlog/bench/formatter-bench.cpp:6:0:
/usr/local/include/benchmark/benchmark.h: In instantiation of 'benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...)::<lambda(benchmark::State&)> [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]':
/usr/local/include/benchmark/benchmark.h:1050:48:   required from 'struct benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...) [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]::<lambda(class benchmark::State&)>'
/usr/local/include/benchmark/benchmark.h:1049:38:   required from 'benchmark::internal::Benchmark* benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...) [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]'
/tmp/tmp.CP8z4tf0ew/vendor/spdlog/bench/formatter-bench.cpp:49:79:   required from here
/usr/local/include/benchmark/benchmark.h:1050:41: error: variable 'fn' has function type
       name, [=](benchmark::State& st) { fn(st, args...); });
                                         ^
compilation terminated due to -Wfatal-errors.
bench/CMakeFiles/formatter-bench.dir/build.make:62: recipe for target 'bench/CMakeFiles/formatter-bench.dir/formatter-bench.cpp.o' failed
make[2]: *** [bench/CMakeFiles/formatter-bench.dir/formatter-bench.cpp.o] Error 1
CMakeFiles/Makefile2:1168: recipe for target 'bench/CMakeFiles/formatter-bench.dir/all' failed
make[1]: *** [bench/CMakeFiles/formatter-bench.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

Followed the documentation and got this.

```
cmake -D CMAKE_CXX_COMPILER=g++-5 -D SPDLOG_BUILD_BENCH=ON -D CMAKE_PREFIX_PATH=/home/mate/benchmark/install ..
-- The CXX compiler identification is GNU 5.5.0
-- Check for working CXX compiler: /usr/bin/g++-5
-- Check for working CXX compiler: /usr/bin/g++-5 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build spdlog: 1.8.2
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Build type: Release
-- Generating example(s)
-- Generating benchmarks
-- Generating install
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mate/spdlog/build
mate@MATTY-GL702ZC:~/spdlog/build$ grep RegisterBenchmark ../bench/formatter-bench.cpp
        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
        //        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
        //        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern)->Iterations(2500000);
        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
mate@MATTY-GL702ZC:~/spdlog/build$ cmake --build . --target formatter-bench | grep error
In file included from /home/mate/spdlog/bench/formatter-bench.cpp:6:0:
/home/mate/benchmark/install/include/benchmark/benchmark.h: In instantiation of ‘benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...)::<lambda(benchmark::State&)> [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]’:
/home/mate/benchmark/install/include/benchmark/benchmark.h:1062:48:   required from ‘struct benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...) [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]::<lambda(class benchmark::State&)>’
/home/mate/benchmark/install/include/benchmark/benchmark.h:1061:38:   required from ‘benchmark::internal::Benchmark* benchmark::RegisterBenchmark(const char*, Lambda&&, Args&& ...) [with Lambda = void (&)(benchmark::State&, std::__cxx11::basic_string<char>); Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]’
/home/mate/spdlog/bench/formatter-bench.cpp:37:79:   required from here
/home/mate/benchmark/install/include/benchmark/benchmark.h:1062:41: error: variable ‘fn’ has function type
       name, [=](benchmark::State& st) { fn(st, args...); });
mate@MATTY-GL702ZC:~/spdlog/build$ grep RegisterBenchmark ../bench/formatter-bench.cpp
        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern);
        //        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
        //        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern)->Iterations(2500000);
        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern);
mate@MATTY-GL702ZC:~/spdlog/build$ cmake --build . --target formatter-bench
[ 80%] Built target spdlog
Scanning dependencies of target formatter-bench
[ 90%] Building CXX object bench/CMakeFiles/formatter-bench.dir/formatter-bench.cpp.o
[100%] Linking CXX executable formatter-bench
[100%] Built target formatter-bench
```

--- a/bench/formatter-bench.cpp
+++ b/bench/formatter-bench.cpp
@@ -34,14 +34,14 @@ void bench_formatters()
     for (auto &flag : all_flags)
     {
         auto pattern = std::string("%") + flag;
-        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
+        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern);
 
         //        pattern = std::string("%16") + flag;
-        //        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
+        //        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern);
         //
         //        // bench center padding
         //        pattern = std::string("%=16") + flag;
-        //        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
+        //        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern);
     }
 
     // complex patterns
@@ -52,7 +52,7 @@ void bench_formatters()
     };
     for (auto &pattern : patterns)
     {
-        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern)->Iterations(2500000);
+        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern)->Iterations(2500000);
     }
 }
 
@@ -73,7 +73,7 @@ int main(int argc, char *argv[])
     }
     else
     {
-        benchmark::RegisterBenchmark(pattern.c_str(), bench_formatter, pattern);
+        benchmark::RegisterBenchmark(pattern.c_str(), &bench_formatter, pattern);
     }
     benchmark::Initialize(&argc, argv);
     benchmark::RunSpecifiedBenchmarks();

